### ISSUE TITLE
bug in reported order of ICC by_group 

### ISSUE THREAD

**Issue by maperez1976** [easystats/performance#154]
Hi,
It seems to be a bug in the ICC reported by groups. I have tried two simple models with only 2 random-effects, which terms are included in different order:

> m1 = lmer(TR_log_Scaled ~ (1|Participant) + (1|item), data = verb_LargeSet_Stand)
> m2 = lmer(TR_log_Scaled ~ (1|item) + (1|Participant), data = verb_LargeSet_Stand)

When ICC by groups are requested, the order of items in the group column corresponds to that in the model, but ICC values are the same for both outputs.

> icc(m1, by_group = TRUE)
# ICC by Group

Group       |   ICC
-------------------
Participant | 0.121
item        | 0.405
> icc(m2, by_group = TRUE)
# ICC by Group

Group       |   ICC
-------------------
item        | 0.121
Participant | 0.405

Regards,
Miguel

**Comment by strengejacke** [comment]
Do you have a reproducible example? For me, it works as intended:

``` r
library(performance)
library(lme4)
#> Warning: Paket 'lme4' wurde unter R Version 4.0.3 erstellt
#> Lade nötiges Paket: Matrix
data(sleepstudy)
set.seed(12345)
sleepstudy$grp <- sample(1:5, size = 180, replace = TRUE)
m1 <- lmer(
  Reaction ~ Days + (1 | grp) + (1 | Subject),
  data = sleepstudy
)
m2 <- lmer(
  Reaction ~ Days + (1 | Subject) + (1 | grp),
  data = sleepstudy
)
icc(m1, by_group = TRUE)
#> # ICC by Group
#> 
#> Group   |   ICC
#> ---------------
#> Subject | 0.590
#> grp     | 0.002
icc(m2, by_group = TRUE)
#> # ICC by Group
#> 
#> Group   |   ICC
#> ---------------
#> Subject | 0.590
#> grp     | 0.002
```

<sup>Created on 2020-11-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by maperez1976** [comment]
Hi,
I cannot share my own data but I created a simple dataset (see [dataset_icc.xlsx](https://github.com/easystats/performance/files/5476410/dataset_icc.xlsx) and repeated the procedure:

> m1 = lmer(RT ~ (1|Participant) + (1|Item), data = dataset_icc)
> m2 = lmer(RT ~ (1|Item) + (1|Participant), data = dataset_icc)
> icc(m1, by_group = TRUE)
# ICC by Group

Group       |   ICC
-------------------
Participant | 0.896
Item        | 0.004
> icc(m2, by_group = TRUE)
# ICC by Group

Group       |   ICC
-------------------
Item        | 0.896
Participant | 0.004

**Comment by strengejacke** [comment]
Can't reproduce, either:

``` r
library(readxl)
dataset_icc <- read_excel("D:/Downloads/dataset_icc.xlsx")

library(lme4)
#> Warning: Paket 'lme4' wurde unter R Version 4.0.3 erstellt
#> Lade nötiges Paket: Matrix
library(performance)
m1 <- lmer(RT ~ (1|Participant) + (1|Item), data = dataset_icc)
m2 <- lmer(RT ~ (1|Item) + (1|Participant), data = dataset_icc)
icc(m1, by_group = TRUE)
#> # ICC by Group
#> 
#> Group       |   ICC
#> -------------------
#> Item        | 0.896
#> Participant | 0.004
icc(m2, by_group = TRUE)
#> # ICC by Group
#> 
#> Group       |   ICC
#> -------------------
#> Item        | 0.896
#> Participant | 0.004
```

<sup>Created on 2020-11-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Are your packages up to date?

**Comment by strengejacke** [comment]
I _think_ your issue was fixed here: https://github.com/easystats/performance/issues/135

**Comment by maperez1976** [comment]
You are right. Error fixed with v0.5.1.
I did not realized that there was an update because I installed the performace package  last week for the first time. Sorry for wasting your time.
Congrats for the package!
Miguel

**Comment by strengejacke** [comment]
Don't worry, glad that it works!
